### ISSUE TITLE
Entry parent is empty when editing after update to 3.7.19

### ISSUE THREAD

**Issue by michaelhue** [craftcms/cms#10050]
### Description

When opening structured entries that have a parent in the control panel, the parent field is empty.
When saving the entry without any changes, the entry is no longer a child and moved to the root level.
When re-selecting the parent entry and saving, the field will be empty again the next time the entry is edited.

This happened after updating to `3.7.19`.
Reverting to `3.7.18.2` solved the issue.

The section in question has a maximum of two levels.
The propagation method is "Save entries to other sites in the same site group".

### Steps to reproduce

1. Create a structured section.
2. Set propagation method to "Save entries to other sites in the same site group".
3. Create a parent entry.
4. Create a child entry.
5. Open the child entry again, check the "Parent" field.

### Additional info

- Craft version: Craft Pro 3.7.19
- PHP version: 8.0.2
- Database driver & version: MySQL 8.0.23

**Comment by brandonkelly** [comment]
Hi @michaelhue, please try doing the following:

1. Create a database backup
2. Re-update to 3.7.19
3. Run the command:
   ```bash
   php craft utils/repair/section-structure <sectionHandle> > repair-output.txt
   ```
4. Once it’s done, search `repair-output.txt` for the entry title that was missing its parent. Is it being inserted at the expected level? If not, what’s the message?

**Comment by michaelhue** [comment]
Hey @brandonkelly!

I did try that on my local installation, but did not save the output. I remember that the structure looked as expected in the command output, but the behavior when editing an entry was the same. Every entry with a parent is affected by this.

**Comment by brandonkelly** [comment]
Not able to reproduce it on my end, so there may be something else in your install that’s interfering. Can you please email support@craftcms.com with a database backup, plus your `composer.json` and `composer.lock` files?

**Comment by wsydney76** [comment]
Can confirm, also with  "Save entries to other sites in the same site group". 

Structure looks ok on the element index after running the command, but parent entry is empty on the edit page.

This effects also element actions like 'Create as children / before /after'..., the new element is created on the top level.

Well not always. It seems to work when the site group contains the 'primary site' ???

**Comment by brandonkelly** [comment]
> It seems to work when the site group contains the 'primary site'

Aha, that was it. I was able to reproduce if the parent entry didn’t belong to the primary site.

Just released Craft 3.7.19.1 with a fix.
